lib/page/assets/index.dart

```diff
@@ -295,15 +301,20 @@ class _AssetsState extends State<Assets> {
       context: context,
       builder: (_) {
         return WillPopScope(
-          child: PasswordInputDialog(
-            title: Text(I18n.of(context).home['unlock']),
+          child: PasswordInputSwitchDialog(
+            title: Text('Unlock account ${store.account.currentAccount.name} with PIN'),
             account: store.account.currentAccount,
             onOk: (password) {
               setState(() {
                 store.account.setPin(password);
               });
             },
-            onCancel: () => _showPasswordNotEnteredDialog(context),
+            onSwitch: () => {
+              NetworkSelectPage(null, null, _update),
+              Navigator.of(context).pop(),
+              // Navigator.of(context).pushNamed('/network')
+              Navigator.of(context).pushNamed(NetworkSelectPage.route),
```

Comment:
I believe this should be the solution to the update problem.
```suggestion
             await Navigator.of(context).pushNamed(NetworkSelectPage.route);
             setState(() {});
```

Reply:
why await in this case?

Reply:
and why does setState work? we jump to another context, so we should be in the NetworkSelectPage after line 316, no?

Reply:
It is essentially the same as what you have been doing. The `await` makes you wait until you return from the `NetworkSelectPage`. When you return, you call setState to be sure to rebuild the widget **after** potential changes have been performed by the `NetworkSelectPage`

Reply:
doesn't await just wait for Navigator.of(context).pushNamed(NetworkSelectPage.route) to return a value, and all not related operations can continue until the awaited value has been returned? 
it will not come back from NetworkSelectPage in this context, it will only go there

Reply:
No, it waits until the `NetworkSelectPage.route` select page is popped from the top of the stack, which may or may not return a value. Which happens below, and then we come back to this place.

https://github.com/encointer/encointer-wallet-flutter/blob/c3644310e16edd3e86fa00eae8e23a345b77bd5f/lib/page/networkSelectPage.dart#L101

I am not sure if you understand `await` correctly. It depends on what you mean with 'not related operations' :). `await` makes the current thread block until the future in question has completed.